tests/Controller/ProductReviewApiTest.php

```diff
@@ -0,0 +1,359 @@
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        $this->client->request('GET', $this->getReviewListUrl($product));
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'authentication/access_denied_response', Response::HTTP_UNAUTHORIZED);
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_not_allow_to_show_product_review_when_it_does_not_exist()
+    {
+        $productReviewsData = $this->loadFixturesFromFile('resources/product_reviews.yml');
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        $this->client->request('GET', $this->getReviewListUrl($product) . '0', [], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'error/not_found_response', Response::HTTP_NOT_FOUND);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_showing_product_review()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $productReviewsData = $this->loadFixturesFromFile('resources/product_reviews.yml');
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        /** @var ReviewInterface $productReview */
+        $productReview = $productReviewsData['productReview1'];
+
+        $this->client->request('GET', $this->getReviewUrl($product, $productReview), [], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'product_review/show_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_indexing_product_reviews()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $productReviewsData = $this->loadFixturesFromFile('resources/product_reviews.yml');
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        $this->client->request('GET', $this->getReviewListUrl($product), [], [], static::$authorizedHeaderWithAccept);
+        $response = $this->client->getResponse();
+
+        $this->assertResponse($response, 'product_review/index_response', Response::HTTP_OK);
+    }
+
+    /**
+     * @test
+     */
+    public function it_allows_create_product_review()
+    {
+        $this->loadFixturesFromFile('authentication/api_administrator.yml');
+        $productReviewsData = $this->loadFixturesFromFile('resources/product_reviews.yml');
+
+        /** @var ProductInterface $product */
+        $product = $productReviewsData['product1'];
+
+        $data =
+<<<EOT
+        {
+          "title": "J_REVIEW",
+          "rating": "3",
+          "comment": "J_REVIEW_COMMENT",
```

Comment:
Don't be afraid of using some more natural exemplary data, these are of course not wrong, but less natural to read. Tests are good place to place easter eggs in code (look at our Behat scenarios and I'll see what I mean 😄 )